### PR TITLE
Ucd sonic

### PR DESCRIPTION
**Pull request by brucemoran**
---
name: UCD_SONIC
about: University College Dublin Sonic HPC config
---

Please follow these steps before submitting your PR:

- [x] If your PR is a work in progress, include `[WIP]` in its title
- [x] Your PR targets the `master` branch
- [x] You've included links to relevant issues, if any

Steps for adding a new config profile:

- [x] Add your custom config file to the `conf/` directory
- [x] Add your documentation file to the `docs/` directory
- [x] Add your custom profile to the `nfcore_custom.config` file in the top-level directory
- [x] Add your custom profile to the `README.md` file in the top-level directory
- [x] Add your profile name to the `profile:` scope in `.github/workflows/main.yml`

Working on my end, let me know if anything else required, thanks!

### CODE REVIEW

File: conf/ucd_sonic.config [hunk 1]
```diff
@@ -0,0 +1,38 @@
+//UCD_SONIC config
```
Comment (jfy133) [inline conf/ucd_sonic.config:1]:
```suggestion
```

File: conf/ucd_sonic.config [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+params {
+  config_profile_name = 'UCD_SONIC'
+  config_profile_description = 'University College Dublin Sonic HPC profile provided by nf-core/configs.'
+  config_profile_contact = 'Bruce Moran (@brucemoran)'
+  config_profile_url = 'https://www.ucd.ie/itservices/ourservices/researchit/researchcomputing/sonichpc/'
+  //max_memory = 1.MB //yes really, allocation made based on other factors and cannot be done via params.max_memory...
```
Comment (jfy133) [inline conf/ucd_sonic.config:7]: This is not an allocation but rather a cap which needs to be defined at a pipeline level - setting to 1.MB will fail! I guess you may have worked that our so I would remoce teh commend

File: conf/ucd_sonic.config [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+process {
+  executor = 'slurm'
+  queue = 'shared'
+  queueSize = 50
+  submitRateLimit = '10 sec'
+  maxRetries = 2
+  beforeScript = 'export NXF_OPTS="-Xms2G -Xmx40G"; module load nextflow/22.04.5.5708 singularity/3.5.2'
+  clusterOptions = { "--mem 1M" } 
+  cache = 'lenient'
+  memory = 1.MB
```
Comment (jfy133) [inline conf/ucd_sonic.config:21]: @maxulysse would this be a problem with inheritance, no?

File: conf/ucd_sonic.config [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+cleanup = true
+
+profiles {
+  moranb {
```
Comment (jfy133) [inline conf/ucd_sonic.config:27]: What is this exactly? Is it for a specific lab? This should be documented

File: docs/ucd_sonic.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# nf-core/configs: UCD Sonic Configuration
+
+The nf-core pipeline [sarek](https://nf-co.re/sarek) has been successfully tested on [Sonic HPC] (https://www.ucd.ie/itservices/ourservices/researchit/researchcomputing/sonichpc) using this config.
+
+This Sonic config offers very minimal options to users but should allow the use of all nf-core pipelines.
+
+## sabtch_nxf_creator
```
Comment (jfy133) [inline docs/ucd_sonic.md:7]:
```suggestion
## sabtch_nxf_creator

```

File: docs/ucd_sonic.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+To run a pipeline, you should use the [sbatch_nxf_creator](https://github.com/brucemoran/sbatch_nxf_creator) method.
```
Comment (jfy133) [inline docs/ucd_sonic.md:8]: This seems rather specific/possibly not generalisable to other users and the way they want to run nexfllow.., is it really necessary? Why would you want to process individually for example and enforce a specific file structure

I would rather describe line 16, and then say you offer a helper script if one wants to do specific what you do.

**Review by jfy133 — CHANGES_REQUESTED** [review]

**Comment by maxulysse** [comment]
@nf-core-bot fix linting pretty please 🙏

File: conf/ucd_sonic.config (hunk 2 above)
Comment (brucemoran) [reply conf/ucd_sonic.config:7]: OK, my mistake

**Review by brucemoran — COMMENTED** [review]
All OK

File: conf/ucd_sonic.config (hunk 3 above)
Comment (maxulysse) [reply conf/ucd_sonic.config:20]: And that sounds a bit too small as well

File: conf/ucd_sonic.config (hunk 2 above)
Comment (jfy133) [reply conf/ucd_sonic.config:6]:
```suggestion
```

File: conf/ucd_sonic.config (hunk 3 above)
Comment (jfy133) [reply conf/ucd_sonic.config:20]: @brucemoran remove here too?

File: docs/ucd_sonic.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
+# nf-core/configs: UCD Sonic Configuration
+
+The nf-core pipeline [sarek](https://nf-co.re/sarek) has been successfully tested on [Sonic HPC] (https://www.ucd.ie/itservices/ourservices/researchit/researchcomputing/sonichpc) using this config.
+
+This Sonic config offers very minimal options to users but should allow the use of all nf-core pipelines.
+
+## sbatch_nxf_creator
+
+To run a pipeline, you can use the [sbatch_nxf_creator](https://github.com/brucemoran/sbatch_nxf_creator) method.
```
Comment (jfy133) [inline docs/ucd_sonic.md:9]:
```suggestion
To run a pipeline, you can optionally use the [sbatch_nxf_creator](https://github.com/brucemoran/sbatch_nxf_creator) method.
```

**Review by jfy133 — COMMENTED** [review]
Almost there! Outstranding questions though

File: conf/ucd_sonic.config (hunk 4 above)
Comment (brucemoran) [reply conf/ucd_sonic.config]: Removed, yes it was for a specific lab.

**Review by jfy133 — APPROVED** [review]

File: conf/ucd_sonic.config [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+process {
+  executor = 'slurm'
+  queue = 'shared'
+  queueSize = 50
+  submitRateLimit = '10 sec'
+  maxRetries = 2
+  beforeScript = 'export NXF_OPTS="-Xms2G -Xmx40G"; module load nextflow/22.04.5.5708 singularity/3.5.2'
+  clusterOptions = { "--mem 1M" }
```
[the remark this answers is not in the document]
Comment (brucemoran) [reply conf/ucd_sonic.config:17]: Sysadmins told me it is the only way it can work, I couldn't believe it but it does seem to be OK!

**Pull request merged**